### DISCUSSION TITLE
Time for v69.2.0?

### DISCUSSION THREAD

**Discussion by abravalheri** [pypa/setuptools#4276] [Dev]
Thanks to the amazing work done by @Avasam recently, the codebase for setuptools has been changing rapidly.

I propose that before merging more refactoring (or any other more serious code change) we consolidate the work that has been done so far in release 69.2.0.

The advantage of splitting the PRs among multiple releases is that, it makes it easier to revert/find bugs if they happen.

Would that be fine for you @jaraco, @Avasam?

If that is OK, we can do it in the next few days (this week is good for me, but the 2 last weeks of March I will not be available to help).

**Comment by Avasam** [comment]
Up to you, I'm slowly getting more and more into work that *could* break something if not cautious. It's also why I've kept runtime changes separate from annotation changes as much as possible.

Currently I'm blocked on https://github.com/pypa/setuptools/pull/4246, which is dependant on https://github.com/pypa/setuptools/pull/4262 .
After that I'm tackling return types, which are likely gonna reveal more unsound code (typing-wise). And once that's done, `pkg_resources` will be on-par with typeshed and can be shipped as "typed" (with a `py.typed` marker). Which would align nicely with a potential 69.3 release.
And after *that*, it's setuptool's turn, using everything we learned from previous PRs ^^

Btw thank you for all your support on this endeavour!

**Reply by abravalheri** [reply]
Thanks @Avasam, unless I hear any concerns about v69.2.0, I am planning on releasing it tomorrow.

The idea is to release incremental updates frequently and avoid bulk updates (as per https://setuptools.pypa.io/en/latest/development/releases.html).

About the #4262, I will have a look shortly after the release.

**Reply by Avasam** [reply]
When you have time for reviews, I'm still blocked on these.

**Comment by abravalheri** [comment]
I have just released v69.2.0 to consolidate all the changes already merged.

Let's see if we have any regressions reported, than we can proceed with further changes.

**Comment by jaraco** [comment]
Good call. Always good to have frequent releases. I do have some distutils changes that need to be merged, but I'd prefer to release those in a separate release anyway, so happy to see the release made.
